phantazm-server/src/main/java/com/github/phantazmnetwork/server/EquipmentFeature.java

```diff
@@ -0,0 +1,505 @@
+import com.github.phantazmnetwork.zombies.equipment.gun.target.tester.PhantazmTargetTester;
+import com.github.phantazmnetwork.zombies.equipment.gun.target.tester.TargetTester;
+import com.github.phantazmnetwork.zombies.equipment.gun.visual.ClipStackMapper;
+import com.github.phantazmnetwork.zombies.equipment.gun.visual.GunStackMapper;
+import com.github.phantazmnetwork.zombies.equipment.gun.visual.ReloadStackMapper;
+import com.github.steanky.ethylene.core.bridge.ConfigBridges;
+import com.github.steanky.ethylene.core.codec.ConfigCodec;
+import com.github.steanky.ethylene.core.processor.ConfigProcessException;
+import com.github.steanky.ethylene.core.processor.ConfigProcessor;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.key.Keyed;
+import net.minestom.server.event.Event;
+import net.minestom.server.event.EventNode;
+import net.minestom.server.event.entity.projectile.ProjectileCollideWithBlockEvent;
+import net.minestom.server.event.entity.projectile.ProjectileCollideWithEntityEvent;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.stream.Stream;
+
+final class EquipmentFeature {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EquipmentFeature.class);
+
+    private static Map<Key, List<ComplexData>> gunLevelMap = null;
+
+    private EquipmentFeature() {
+        throw new UnsupportedOperationException();
+    }
+
+    static void initialize(@NotNull Path equipmentPath, @NotNull ConfigCodec codec) {
+        String ending;
+        if (codec.getPreferredExtensions().isEmpty()) {
+            ending = "";
+        }
+        else {
+            ending = "." + codec.getPreferredExtensions().get(0);
+        }
+        PathMatcher matcher = equipmentPath.getFileSystem().getPathMatcher("glob:**" + ending);
+
+        Path guns = equipmentPath.resolve("guns");
+        try {
+            Files.createDirectories(guns);
+        } catch (IOException e) {
+            LOGGER.warn("Failed to create guns directory.", e);
+            return;
+        }
+
+        ConfigProcessor<GunData> gunDataProcessor = GunData.processor();
+        ConfigProcessor<ComplexData> gunLevelProcessor = createGunLevelProcessor();
+        Map<Key, BiConsumer<? extends Keyed, Collection<Key>>> dependencyAdders = createDependencyAdders();
+
+        gunLevelMap = new HashMap<>();
+        try (Stream<Path> gunDirectories = Files.list(guns)) {
+            gunIteration: for (Path gunDirectory : (Iterable<? extends Path>) gunDirectories::iterator) {
+                if (!Files.isDirectory(gunDirectory)) {
```

Comment:
you should be able to exclude directories by passing a predicate to Files.list

Reply:
the ones with maxDepth? do I set it to 1?

Reply:
yes and yes